substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/io/Utf8.java

```diff
@@ -36,8 +36,22 @@ public final class Utf8 {
     private Utf8() {
     }
 
+    private static int utf8Size(char c) {
+        // Based On
+        // https://github.com/openjdk/jdk21u/blob/jdk-21.0.6%2B7/src/hotspot/share/utilities/utf8.cpp#L409-L418
```

Comment:
JDK-8314794 is new in 21.0.4. So should be that release.

```suggestion
        // https://github.com/openjdk/jdk21u/blob/jdk-21.0.4%2B7/src/hotspot/share/utilities/utf8.cpp#L409-L418
```

Reply:
I see that @luneo7 already patched this (which I am OK with), but I will comment for future reference:

I believe that's not that important. AFAIU the intention of these annotations is to highlight the code segment used/copied at the time of creation of this snippet, i.e. it's not important when the corresponding upstream code first appeared.

FWIW the code being copied here was first introduced 9 years ago (in jdk-9+146) by https://github.com/openjdk/jdk21u/commit/ed9adcbdfa82e2e8b08b69577c53c94207909c2a#diff-9773cdf9015077680910358eb0a75968c000012f19d70e227829fa3e024196e2R414-R421, I don't think we should reference that tag though.